Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift

```diff
@@ -2232,419 +2236,77 @@ class PaymentSheetDeferredServerSideUITests: PaymentSheetUITestCase {
- waitForExpectations(timeout: 10, handler: nil)
- payButton.tap()
- 
- let successText = app.alerts.staticTexts["Success!"]
- XCTAssertTrue(successText.waitForExistence(timeout: 10))
- 
- let okButton = app.alerts.buttons["OK"]
- okButton.tap()
- }
- 
- private func linkLogin() throws {
- let modal = app.otherElements["Stripe.Link.PayWithLinkWebController"]
- XCTAssertTrue(modal.waitForExistence(timeout: 10))
- 
- let emailField = modal.textFields["Email"]
- XCTAssertTrue(emailField.waitForExistence(timeout: 10))
- emailField.tap()
- emailField.typeText("mobile-payments-sdk-ci+a-consumer@stripe.com")
- 
- // Wait for OTP screen and enter the code
- let codeField = app.descendants(matching: .any)["Code field"]
- XCTAssert(codeField.waitForExistence(timeout: 10))
- codeField.tap()
- app.typeTextWithKeyboard("000000")
- }
- 
- private func loginAndPay() throws {
- try linkLogin()
- 
- let modal = app.otherElements["Stripe.Link.PayWithLinkWebController"]
- let paymentMethodPicker = app.otherElements["Stripe.Link.PaymentMethodPicker"]
- if paymentMethodPicker.waitForExistence(timeout: 10) {
- paymentMethodPicker.tap()
- paymentMethodPicker.buttons["Add a payment method"].tap()
- }
- 
- try fillCardData(app, container: modal)
- 
- // Pay!
- let payButton = modal.buttons["Pay $50.99"]
- expectation(for: NSPredicate(format: "enabled == true"), evaluatedWith: payButton, handler: nil)
- waitForExpectations(timeout: 10, handler: nil)
- payButton.tap()
- 
- let successText = app.alerts.staticTexts["Success!"]
- XCTAssertTrue(successText.waitForExistence(timeout: 10))
- 
- let okButton = app.alerts.buttons["OK"]
- okButton.tap()
- }
- }
- */
+class PaymentSheetLinkUITests: PaymentSheetUITestCase {
+    // MARK: Inline signup
+    /// Tests the Link inline signup flow.
+    func testLinkInlineSignup() throws {
+        var settings = PaymentSheetTestPlaygroundSettings.defaultValues()
+        settings.customerMode = .guest
+        settings.apmsEnabled = .on
+        settings.linkEnabled = .on
+        settings.linkV2Allowed = .on
+
+        loadPlayground(app, settings)
+
+        app.buttons["Present PaymentSheet"].tap()
+
+        try fillCardData(app)
+
+        app.switches["Save your info for secure 1-click checkout with Link"].tap()
+
+        let emailField = app.textFields["Email"]
+        emailField.tap()
+        emailField.typeText("mobile-payments-sdk-ci+\(UUID())@stripe.com")
+
+        let phoneField = app.textFields["Phone"]
+        // Phone field appears after the network call finishes. We want to wait for it to appear.
+        XCTAssert(phoneField.waitForExistence(timeout: 10))
+        phoneField.tap()
+        phoneField.typeText("3105551234")
+
+        // The name field is only required for non-US countries. Only fill it out if it exists.
+        let nameField = app.textFields["Name"]
+        if nameField.exists {
+            nameField.tap()
+            nameField.typeText("Jane Done")
+        }
```

Comment:
Seems like we should have two tests here:
* One test without a name field (US country)
* One test with a name field (non-us country)

Reply:
MOBILESDK-1476